### PR TITLE
Make UserAgent a singleton

### PR DESCRIPTION
**Pull request by hichamboushaba**
### Description
While helping the investigation of peaMlT-Tk-p2, I noticed an issue with our DI configuration, the `UserAgent` class is not singleton, and given it has a background initialization for a relatively heavy operation, instantiating multiple times is problematic:
- It's not great for the app performance.
- It increases the chance of using the [default](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/07f08124c02b03d5d380cdaf7066ec4ccf9457f4/fluxc/src/main/java/org/wordpress/android/fluxc/network/UserAgent.kt#L26) user agent, as we will be using a fresh instance instead of the cached one.

### Testing information
This shouldn't have any impact, testing some screens like products and orders to confirm network operations are not impacted would be enough.

### The tests that have been performed
I added a log entry to the `provideUserAgent` function to check how many times it's executed, and confirmed that the `Singleton` annotation works as expected.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/237">20.5 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/di/AppConfigModule.kt [hunk 1]
```diff
@@ -25,6 +25,7 @@ class AppConfigModule {
     fun provideAppSecrets() = AppSecrets(BuildConfig.OAUTH_APP_ID, BuildConfig.OAUTH_APP_SECRET)
 
     @Provides
+    @Singleton
     fun provideUserAgent(appContext: Context) = UserAgent(appContext, USER_AGENT_APPNAME)
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/di/AppConfigModule.kt:29]: Shell, we do the same for `com. woo.commerce.android.wear.diAppConfigModule.kt`?

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/di/AppConfigModule.kt:29]: Yes, nice catch, I added in the last commit d7eefc8501829d96691ec4ffd7127a0c6bc9a98b

**Review by kidinov — APPROVED** [review]
LGTM. Validated using debugger - initialised once

**Pull request merged**